bluepysnap/circuit_validation.py

```diff
@@ -740,11 +740,117 @@ def validate(config_file, bbp_check=False):
     Returns:
         list: List of errors, empty if no errors
     """
-    config = Config(config_file).resolve()
+    config = Config(config_file)
+    if bbp_check:
+        validate_schemas(config)
+
+    config = config.resolve()
+
     errors = _check_required_datasets(config)
     if not errors:
         errors = _check_populations(config)
     if not bbp_check:
         errors = [e for e in errors if not isinstance(e, BbpError)]
     _print_errors(errors)
     return set(errors)
+
+
+def get_h5_structure_as_dict(h5):
+    """Recursively translates h5 file into a dictionary.
+
+    For groups, the subgroups/datasets are translated as a subdictionary.
+    Datatype of datasets is resolved and returned as {'datatype': <dtype>}.
+    Attributes of either groups or datasets are returned as {'attributes': {<key>: <value>}}.
+
+    Args:
+        h5 (h5.File instance): h5 file to translate
+
+    Returns:
+        dict: dictionary of the file structure
+    """
+    properties = {}
+
+    def get_dataset_dtype(item):
+        if item.dtype.hasobject:
+            return h5py.check_string_dtype(item.dtype).encoding
+
+        return item.dtype.name
+
+    for key, value in h5.items():
+        if isinstance(value, h5py.Group):
+            properties[key] = get_h5_structure_as_dict(value)
+        elif isinstance(value, h5py.Dataset):
+            properties[key] = {'datatype': get_dataset_dtype(value)}
+        else:
+            properties[key] = {}
+
+        attrs = {k: v for k, v in value.attrs.items()}
+        if attrs:
+            properties[key]['attributes'] = attrs
+
+    # Maybe check for the existence of @library and do the translation to actual values here?
```

Comment:
yeah, that would be a good idea; perhaps it's worth also noting that they come from `@library`, in case we want to verify that too